fh-sync.d.ts

```diff
@@ -0,0 +1,350 @@
+// Type definitions for fh-sync
+// Project: https://github.com/feedhenry/fh-sync
+// Maintainer feedhenry-dev@redhat.com
+
+declare module SyncCloud {
+
+      /**
+       * Valid actions (the "fn" param) that can be passed to sync.invoke
+       */
+      type InvokeAction = 'sync'|'syncRecords'|'listCollisions'|'removeCollision'
+
+      /**
+       * Options that can be passed to sync.invoke
+       */
+      interface InvokeOptions {
+          fn: InvokeAction
+
+          // TODO: we should define this in more detail
+          [key: string]: any
+      }
+
+      /**
+       * Interfaces that describe how sync responses should be structured
+       */
+      namespace HandlerResults {
```

Comment:
By having these defined and used by the callbacks for handlers we can have TS notify a developer that they're returning the correct/incorrect data structure.

For example:

```ts
sync.handleList('items', (dataset, query, meta, done) => {
  done(null, [1,2,3]) // this will not compile with the new changes
  done(null, { 431321: { /* data goes here */ } }) // this will compile since it's a hash
})
```